### ISSUE TITLE
Implement PartShop.downloadAttachment

### ISSUE THREAD

**Issue by JoshuaUrrutia** [SynBioDex/pySBOL2#364]
The pull method doesn't seem to work on attachments, when I try:
```
sd2_sbh.pull('https://hub.sd2e.org/user/sd2e/UCSB_GBW_LandingSiteDesignv0_1through36/attachment_e78e402799cb429d9fafa1c9dcb36ab4/1', doc)
```
I get back an error:
```
---------------------------------------------------------------------------
SBOLError                                 Traceback (most recent call last)
<ipython-input-33-6d32e39c7e97> in <module>
----> 1 sd2_sbh.pull('https://hub.sd2e.org/user/sd2e/UCSB_GBW_LandingSiteDesignv0_1through36/attachment_e78e402799cb429d9fafa1c9dcb36ab4/1', doc)
      2 #'https://hub.sd2e.org/user/sd2e/UCSB_GBW_LandingSiteDesignv0_1through36/attachment_e78e402799cb429d9fafa1c9dcb36ab4/1/sbol'
~/tacc-work/jupyter_packages/lib/python3.6/site-packages/sbol2/partshop.py in pull(self, uris, doc, recursive)
    151             if response.status_code == 404:
    152                 raise SBOLError(SBOLErrorCode.SBOL_ERROR_NOT_FOUND,
--> 153                                 'Part not found. Unable to pull: ' + query)
    154             elif response.status_code == 401:
    155                 raise SBOLError(SBOLErrorCode.SBOL_ERROR_HTTP_UNAUTHORIZED,
SBOLError: (<SBOLErrorCode.SBOL_ERROR_NOT_FOUND: 3>, 'Part not found. Unable to pull: https://hub.sd2e.org/user/sd2e/UCSB_GBW_LandingSiteDesignv0_1through36/attachment_e78e402799cb429d9fafa1c9dcb36ab4/1/sbol')
```
But downloading the sbol through the GUI:
https://hub.sd2e.org/user/sd2e/UCSB_GBW_LandingSiteDesignv0_1through36/attachment_e78e402799cb429d9fafa1c9dcb36ab4/1
seems to have the same url:
https://hub.sd2e.org/user/sd2e/UCSB_GBW_LandingSiteDesignv0_1through36/attachment_e78e402799cb429d9fafa1c9dcb36ab4/1/sbol

And this same method seems to work fine for sequence uris.

**Comment by JoshuaUrrutia** [comment]
And this is how I created the `sbh_stage` object:
```
import sbol2
SBH_USER = 'username'
SBH_PASSWORD = 'password'
sbh_stage = sbol2.PartShop('https://hub-staging.sd2e.org')
sbh_stage.spoof("https://hub.sd2e.org")
sbh_stage.login(SBH_USER, SBH_PASSWORD)
```

**Comment by tcmitchell** [comment]
The original issue appears to be a problem with data being different on a staging instance of SynBioHub than on the production instance. If you go to the attachment URL you go to the production instance, but if you follow the URL via the staging instance plus spoofing, the attachment is genuinely not there.

`PartShop.pull()` works against the production instance just fine. The catch is that what you download is SBOL _describing_ the attachment, not the attachment itself. pySBOL has a `PartShop.downloadAttachment(attachment_uri, destination_path)` method to download attachments. pySBOL2 is lacking that method, so it should be added.

Once we have the `downloadAttachment` method it would then be possible to download the attached GFF file (the format of this particular attachment), and then [Document.importFromFormat](#329) (once it is implemented) to get the contents of the GFF attachment into an SBOL document (if that's what is desired).

Long story short, implement `PartShop.downloadAttachment(attachment_uri, destination_path)` for this issue. I'll retitle this issue accordingly.

**Comment by tcmitchell** [comment]
Closed by #377
